### DISCUSSION TITLE
Collaborating with non-LaTeX savy advisers

### DISCUSSION THREAD

**Discussion by GilOliveira** [joaomlourenco/novathesis#319] [Q&A] [answered]
I've been faced with a dilemma that I figured should be relatively common among people in this community which is... **how to collaborate with an adviser that does not know LaTeX or git**.

PDF comments were out of the question, at least in my particular case, because they did not offer the flexibility for the adviser to propose substantial changes in the text and move things around.

Interestingly, I was not the only one in my lab who'd been faced with the loyal MS Word professor, and, the best (yet imperfect) solution I found is to compile to PDF, then use a very specific piece of software called [PDF-Xchange](https://www.tracker-software.com/product/pdf-xchange-editor) (commercial, paid) to convert it to `.docx`. It's by all means an imperfect solution, as it required me to eliminate all the plots and covers of my thesis from the PDF before conversion, as the software tries to convert all of the complex graphs to shapes and absolutely destroy MS Word's ability to open the document without hanging.

In any case that's the best way I found and should anyone here be having the same problem, I hope it helps.

My question to you all is: what is the best way you found to collaborate with an adviser that doesn't know LaTeX? And, is there a reliable way to get a simple `.docx` version of the document that we can use so that our advisers can mess all they want with our document, and we just manually bring their feedback back to LaTeX? Thanks!

_P.S.: Please avoid the "just use Word" solution. Yes, this can be an issue, but LaTeX still gives me many advantages over the Microsoft stuff._

**Comment by joaomlourenco** [comment]
@GilOliveira, what a nice discussion topic!   Thanks for starting this thread.

My opinion: 

_The thesis/dissertation is a document from the student, not from the adviser.  This means the adviser has the right (and the duty) to advise but does not have the right to edit (add, remove, change) content directly.  That's up to the student to properly incorporate the suggestions given by the adviser.  This means that commenting on the PDF is not only desirable as it should be standard procedure._

Unfortunately, there are lots of advisers who think different and want to edit the student's document and do impose they will. 😪

The most time-effective approach I can devise for a student that has to deal with an adviser that really _wants the document in Word_ is for the student to take some anxiolytic (Xanax, Lexotan, 😊) and really use Word, but without caring for the look and feel of the document.   Once the adviser agrees that the document is stable/concluded in terms of scientific and technical contents, spend a few days converting it to LaTeX.  And you are done with a happy adviser, a beautifully typeset document, and a sane (enough) brain!

But I am seriously wondering how other students deal with this problem!

**Reply by GilOliveira** [reply]
Interesting approach!
As for the major changes from the adviser, of course they shouldn't be done, but PDF annotations are still very limiting for simple changes like reorganizing paragraphs.
Still doing the first drafts in Word isn't the most ideal of solutions because of the advantages LaTeX has even during the writing process (and as someone familiar with anxiety disorders, medication only does so much to help the adviser-advisee relationship).

Maybe I'm too much of an optimist, but I was hopping someone had already made a script that takes care of making the dirty drafts in Word, I've read some interesting approaches, like converting to HTML and then to Word, but I'm still very keen on hearing the folks' approaches to getting feedback...

**Reply by joaomlourenco** [reply]
To reinforce my opinion:

_A shared paper would have (has) your name and your adviser's name.  It is a common work and a shared document.  Both are supposed to work on it somehow.   *Your dissertation* only has your name… you should be made accountable for 100% of it and only you should be the one editing it._

**Comment by b-pereira** [comment, marked as answer]
> I've been faced with a dilemma that I figured should be relatively common among people in this community which is... **how to collaborate with an adviser that does not know LaTeX or git**.
> 
> PDF comments were out of the question, at least in my particular case, because they did not offer the flexibility for the adviser to propose substantial changes in the text and move things around.
> 
> Interestingly, I was not the only one in my lab who'd been faced with the loyal MS Word professor, and, the best (yet imperfect) solution I found is to compile to PDF, then use a very specific piece of software called [PDF-Xchange](https://www.tracker-software.com/product/pdf-xchange-editor) (commercial, paid) to convert it to `.docx`. It's, by all means, an imperfect solution, as it required me to eliminate all the plots and covers of my thesis from the PDF before conversion, as the software tries to convert all of the complex graphs to shapes and destroy MS Word's ability to open the document without hanging.
> 
> In any case that's the best way I found and should anyone here be having the same problem, I hope it helps.
> 
> My question to you all is: what is the best way you found to collaborate with an adviser that doesn't know LaTeX? And, is there a reliable way to get a simple `.docx` version of the document that we can use so that our advisers can mess all they want with our document, and we just manually bring their feedback back to LaTeX? Thanks!
> 
> _P.S.: Please avoid the "just use Word" solution. Yes, this can be an issue, but LaTeX still gives me many advantages over the Microsoft stuff._

I support @joaomlourenco's answer. You should be able to write your document on anything. What matters is the content, not the technology that you use. 

As for PDF-XChange Editor, you have the [PDF-XChange **Viewer**](https://www.tracker-software.com/product/pdf-xchange-viewer) that is the older version of the Editor and you can use it for free without some of the features. I didn't try to use it to convert to Docx but give it a try. 

As for technical issues, an approach is to write your dissertation in Markdown and then use Pandoc to convert to Latex and Word. Of course, programmed graphics don't work and your workflow is already hellish. You can aggregate all Markdown documents in a single file with the `cat` command and then convert to Docx, as also convert to Latex. Is imperfect, but you manage to get the flexibility of Latex (compile 1000 pages document; the formatting that you need because is already programmed in the template - just copy and paste and edit some things; Math environments, images, tables; for citations you can use Zotero and [Better BibTeX for Zotero](https://retorque.re/zotero-better-bibtex/) where you can [drag and drop](https://retorque.re/zotero-better-bibtex/installation/preferences/#quick-copydrag-and-drop-citations) citations from Zotero into your markdown documents and you can [cite as you write](https://retorque.re/zotero-better-bibtex/citing/cayw/) in your favourite editor with varying levels of comfort. Just need to install `pandoc-citeproc`; Check the documentation). Also, you can import and write the Markdown directly to Latex [(see *Writing Markdown in LaTeX Documents*)](https://www.overleaf.com/learn/how-to/Writing_Markdown_in_LaTeX_Documents).

**As you can see this is a *hell* of a workflow. You might see some advantages to this or not. They are just *ideas***.

**Even if any of the workflows work for you, maybe the bottom line is HOW YOUR ADVISER REVIEWS YOUR WORK. I think that the only thing that he complains about is that he has to change something on HIS workflow.** Overleaf offers features to [Track changes and comments in LaTeX](https://www.overleaf.com/track-changes-and-comments-in-latex) and Rich Text viewing for the objection "I can't read  LaTeX code" [(see *Rich Text and Adding Linked Files*)](https://www.overleaf.com/blog/new-overleaf-v2-features-rich-text-and-adding-linked-files). If the argumentation doesn't suffice, leave it be. As Zhuang Zhou put it:

> You cannot speak of ocean to a well-frog, the creature of a narrower sphere. You cannot speak of ice to a summer insect, the creature of a season.

Good luck!

**Reply by GilOliveira** [reply]
Thank you so much @b-pereira, Zotero is already part of my workflow so there would be no changes necessary there, it didn't even came to my mind that I could use Markdown in this case, that does seem very interesting indeed, I'll take a look!
As for Overleaf, it has not, however, not been part of my workflow. I've been using local instances of LaTeX with git because of my (perhaps irrational) fear to leave my precious thesis files 100% on [someone else's computer](https://blog.codinghorror.com/content/images/2019/02/there-is-no-cloud.png). But it may be an interesting platform to just share the current version with my adviser for feedback.

As for your conclusion... well, it's complex, but as is unfortunately the case in some other cases in university, sometimes you got to pick your battles and just do it the faculty's way.

**Reply by joaomlourenco** [reply]
@GilOliveira, please notice that an Overleaf project is also a Git repository, so you may have your local version and work locally.  Then you commit and your adviser may comment in Overleaf.  That sounds as an approach with some potential.

**Reply by GilOliveira** [reply]
@joaomlourenco Was not aware! From the little interactions I had with it it seemed to handle files a bit more like Google Drive. Will check it out then, thanks!

**Reply by joaomlourenco** [reply]
In your Overleaf project, Click on the top-left "Menu", then in the "Sync" section, select "*Git*".
If you select "GitHub" you can make automatic syncing with a project in gitHub.

Some of this features are only available to "Pro" (paid) users.

**Reply by GilOliveira** [reply]
> In your Overleaf project, Click on the top-left "Menu", then in the "Sync" section, select "_Git_". If you select "GitHub" you can make automatic syncing with a project in gitHub.
> 
> Some of this features are only available to "Pro" (paid) users.

Just set it up a few minutes ago, works flawlessly! Thanks!

**Reply by b-pereira** [reply]
> Thank you so much @b-pereira, Zotero is already part of my workflow so there would be no changes necessary there, it didn't even come to my mind that I could use Markdown in this case, that does seem very interesting indeed, I'll take a look! As for Overleaf, it has not, however, not been part of my workflow. I've been using local instances of LaTeX with git because of my (perhaps irrational) fear to leave my precious thesis files 100% on [someone else's computer](https://blog.codinghorror.com/content/images/2019/02/there-is-no-cloud.png). But it may be an interesting platform to just share the current version with my adviser for feedback.
> 
> As for your conclusion... well, it's complex, but as is, unfortunately, the case in some other cases in university, sometimes you got to pick your battles and just do it the faculty's way.

You should read [Jeff Atwood](http://en.wikipedia.org/wiki/Jeff_Atwood)['s *actual* blog post](https://blog.codinghorror.com/the-cloud-is-just-someone-elses-computer/) and you'll get the context of "there is no cloud, it's someone else's computer". 

As per security issues read [Is writing a research paper on Overleaf risky for plagiarism?](https://academia.stackexchange.com/questions/164393/is-writing-a-research-paper-on-overleaf-risky-for-plagiarism), in particular, the comment section. There is ***one*** particular comment that is essential, that you can see below:
![image](https://user-images.githubusercontent.com/944511/158880630-1f61806a-267d-4f66-9ef0-f2eee82bb7a8.png)

The question, all the answers and every comment are worth reading.

**Comment by GilOliveira** [comment]
Thank you so much for all your help! @b-pereira's solution seems to me to be as good as possible so I marked their post as an answer. I do very much welcome more feedback from folks as I'm sure this discussion is bound to help a lot of other students.

Take care!
